nova_vm/src/ecmascript/builtins/number.rs

```diff
@@ -224,4 +265,68 @@ impl NumberConstructor {
         // 5. Set O.[[NumberData]] to n.
         // 6. Return O.
     }
+
+    /// ### [21.1.2.2 Number.isFinite ( number )](https://tc39.es/ecma262/#sec-number.isfinite)
+    fn is_finite(
+        agent: &mut Agent,
+        _this_value: Value,
+        arguments: ArgumentsList,
+    ) -> JsResult<Value> {
+        let maybe_number = arguments.get(0);
+
+        // 1. If number is not a Number, return false.
+        let Ok(number) = Number::try_from(maybe_number) else {
+            return Ok(false.into());
+        };
+
+        // 2. If number is not finite, return false.
+        // 3. Otherwise, return true.
+        Ok(number.is_finite(agent).into())
+    }
+
+    /// ### [21.1.2.3 Number.isInteger ( number )](21.1.2.3 Number.isInteger ( number ))
+    fn is_integer(
+        agent: &mut Agent,
+        _this_value: Value,
+        arguments: ArgumentsList,
+    ) -> JsResult<Value> {
+        let maybe_number = arguments.get(0);
+
+        // 1. Return IsIntegralNumber(number).
+        Ok(is_integral_number(agent, maybe_number).into())
+    }
+
+    /// ### [21.1.2.4 Number.isNaN ( number )](https://tc39.es/ecma262/#sec-number.isnan)
+    fn is_nan(agent: &mut Agent, _this_value: Value, arguments: ArgumentsList) -> JsResult<Value> {
+        let maybe_number = arguments.get(0);
+
+        // 1. If number is not a Number, return false.
+        let Ok(number) = Number::try_from(maybe_number) else {
+            return Ok(false.into());
+        };
+
+        // 2. If number is NaN, return true.
+        // 3. Otherwise, return false.
+        Ok(number.is_nan(agent).into())
+    }
+
+    // ### [21.1.2.5 Number.isSafeInteger ( number )](21.1.2.5 Number.isSafeInteger ( number ))
+    fn is_safe_integer(
+        agent: &mut Agent,
+        _this_value: Value,
+        arguments: ArgumentsList,
+    ) -> JsResult<Value> {
+        let maybe_number = arguments.get(0);
+
+        // 1. If IsIntegralNumber(number) is true, then
+        if is_integral_number(agent, maybe_number) {
```

Comment:
**issue:** This can be optimised into just a `Number::Integer` check. At least until we support 32-bit environments where `Integer` cannot handle the full safe integer range, if we ever do, it should never be the case that any other number is a safe integer.